src/sig_stfl/sig_stfl.h

```diff
@@ -54,6 +54,74 @@ OQS_API int OQS_SIG_STFL_alg_count(void);
  */
 OQS_API int OQS_SIG_STFL_alg_is_enabled(const char *method_name);
 
+/**
+ * @brief OQS_SECRET_KEY object for stateful signature schemes
+ */
+typedef struct OQS_SECRET_KEY OQS_SECRET_KEY;
+
+typedef struct OQS_SECRET_KEY {
+    /** A local ordinal representing the LMS parameter of the signature scheme. */
+    uint32_t oid;
+
+    /** Printable string representing the name of the signature scheme. */
+    const char *method_name;
+
+    /* The (maximum) length, in bytes, of secret keys for this signature scheme. */
+    size_t length_secret_key;
+
+    /* The variant specific secret key data */
+    void *secret_key_data;
+
+    /* User defined data that may be used for the SAFETY functions */
+    void *data;
+
+    /* Function that returns the total number of signatures for the secret key */
+    unsigned long long (*sigs_total)(const OQS_SECRET_KEY *secret_key);
+
+    /* Function that returns the number of signatures left for the secret key */
+    unsigned long long (*sigs_left)(const OQS_SECRET_KEY *secret_key);
+
+    /**
+     * Secret Key retrieval Function
+     *
+     * @param[in] sk The secret key represented as OQS_SECRET_KEY object
+     * @return byte pointer or null
+     */
+    uint8_t * (*get_key)(OQS_SECRET_KEY *sk);
+
+    /**
+     * Secret Key Locking Function
+     *
+     * @param[in] sk The secret key represented as OQS_SECRET_KEY object
+     * @return OQS_SUCCESS or OQS_ERROR
+     */
+    OQS_STATUS (*lock_key)(OQS_SECRET_KEY *sk);
+
+    /**
+     * Secret Key Saving Function
+     *
+     * @param[in] sk The secret key represented as OQS_SECRET_KEY object
+     * @return OQS_SUCCESS or OQS_ERROR
+     */
+    OQS_STATUS (*save_secret_key)(const OQS_SECRET_KEY *sk);
+
+    /**
+     * Secret Key Unlocking / Releasing Function
+     *
+     * @param[in] sk The secret key represented as OQS_SECRET_KEY object
+     * @return OQS_SUCCESS or OQS_ERROR
+     */
+    OQS_STATUS (*release_key)(OQS_SECRET_KEY *sk);
```

Comment:
I know we had `save_secret_key` and `release_key` in the earlier API, but now I'm wondering about the difference between those two.  For a calling application, could it just incorporate whatever action it would take during `save_secret_key` into `release_key`?  When would these ever need to be separated?

Reply:
renamed 'release_key' to 'unlock_key'. I tok 'save_secret_key' to meant to persist the the key (file write).